### PR TITLE
DE-7553: Smarten up the examples repo, organise things a bit

### PR DESCRIPTION
**Pull request by rmoff**

### CODE REVIEW

**Review by hpgrahsl — APPROVED** [review]
Overall LGTM.

File: README.md [hunk 1]
```diff
@@ -1,45 +1,93 @@
 # Decodable Examples
 
+![A squirrel holding a laptop that says "Examples repo"](images/examples_repo.webp)
+
+## Introduction
+
 This repository contains examples of use cases that utilize Decodable streaming solution as well as demos for related open-source projects such as Apache Flink, Debezium, and Postgres.
 
+Examples are presented "as-is" and were working when committed to the repository :)
```
Comment (gunnarmorling) [inline README.md]: How about framing it a bit more positively:

```suggestion
Examples are presented "as-is" and are maintained on a best effort basis. PRs for updating existing (or adding new) examples are welcome!
```

File: README.md [hunk 2]
```diff
@@ -1,45 +1,93 @@
...
+
+### Flink SQL
+
+| Example                                               | Description |
+|-------------------------------------------------------|-------------|
+| [Flink SQL Troubleshooting](troubleshooting-flinksql) | A set of Docker Compose environments for demonstrating various Flink SQL troubleshooting scenarios (see [related blog](https://www.decodable.co/blog/flink-sql-misconfiguration-misunderstanding-and-mishaps?utm_medium=github&utm_source=examples_repo&utm_campaign=blog&utm_content=troubleshooting-flinksql))|
+
+## Deprecated
+
+The following examples are included for reference but are deprecated for the reason shown.
```
Comment (gunnarmorling) [inline README.md]: Is there any value in keeping them, or should we just drop them?

**Review by gunnarmorling — APPROVED** [review]
LGTM. 1.5 suggestions inline.

File: README.md (hunk 2 above)
Comment (rmoff) [reply README.md]: I didn't want to be too aggressive in my pruning. Happy to drop them though. WDYT?

Comment (gunnarmorling) [reply README.md]: I'd drop them.

Comment (rmoff) [reply README.md]: 👍

**Pull request merged**